src/NuGet.Core/NuGet.Protocol/HttpSource/HttpHandlerResourceV3Provider.cs

```diff
@@ -66,7 +66,12 @@ private static HttpHandlerResourceV3 CreateResource(PackageSource packageSource)
 
             if (proxy != null)
             {
-                messageHandler = new ProxyAuthenticationHandler(clientHandler, HttpHandlerResourceV3.CredentialService?.Value, ProxyCache.Instance);
+                var innerHandler = messageHandler;
+
+                messageHandler = new ProxyAuthenticationHandler(clientHandler, HttpHandlerResourceV3.CredentialService?.Value, ProxyCache.Instance)
+                {
+                    InnerHandler = innerHandler
+                };
```

Comment:
Can we add a test, to make sure this doesn't regress in the future?

I'm thinking we add an `internal` constructor (so it doesn't change public APIs, I hate supporting public APIs, it makes product changes so much harder sometimes. Anyway, our test assemblies have InternalsVisibleTo) that takes in a `IProxyCache`, and the existing public constructor calls it with `ProxyCache.Instance`. Then, above on line 42 we call the instance from the constructor.

Therefore, we can write a test that mocks `IWebProxy` to always say a proxy is needed, calls `HttpHandlerResourceV3Provider.CreateResource`, and then assert that the return value's HttpHandler chain contains both ProxyAuthenticationHandler and also ServerWarningLogHandler